### ISSUE TITLE
[FVT]there are error messages when upgrading xcat on ubuntu 

### ISSUE THREAD

**Issue by junxiawang** [xcat2/xcat-core#1057]
env:
ubuntu 16.06 
build:
latest build 
apt-get upgrade xcat
```
Unpacking debootstrap (1.0.78+nmu1ubuntu1.1) over (1.0.78+nmu1ubuntu1) ...
Preparing to unpack .../perl-xcat_2.12-snap201605052044_all.deb ...
Unpacking perl-xcat (2.12-snap201605052044) over (2.12-snap201604040933) ...
Reloading xcatd Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^config \* { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/conserver.pm line 194.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/rmdev -l \${ <-- HERE BASECUST_REMOVAL}/ at /opt/xcat/lib/perl/xCAT_plugin/aixinstall.pm line 7741.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^config \* { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/conserver.pm line 194.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/forwarders { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1063.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/allow-transfer { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1070.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/also-notify { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1077.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1084.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on-v6 { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1091.
 * 
Preparing to unpack .../xcat-client_2.12-snap201605052044_all.deb ...
Unpacking xcat-client (2.12-snap201605052044) over (2.12-snap201604040933) ...
Preparing to unpack .../xcat-server_2.12-snap201605052044_all.deb ...
Unpacking xcat-server (2.12-snap201605052044) over (2.12-snap201604040933) ...
Preparing to unpack .../xcat-genesis-scripts_2.12-snap201605052044_ppc64el.deb ...
Unpacking xcat-genesis-scripts (2.12-snap201605052044) over (2.12-snap201604040933) ...
dpkg: error processing archive /xcatn/xcat-core/pool/main/x/xcat-genesis-scripts/xcat-genesis-scripts_2.12-snap201605052044_ppc64el.deb (--unpack):
 trying to overwrite '/opt/xcat/share/xcat/netboot/genesis/ppc64/fs/etc/udev/rules.d/98-mlx.rules', which is also in package xcat-genesis-base-ppc64 2.11-snap201511160819
Preparing to unpack .../xcat-buildkit_2.12-snap201605052044_all.deb ...
Unpacking xcat-buildkit (2.12-snap201605052044) over (2.12-snap201604040933) ...
Preparing to unpack .../xcat_2.12-snap201605052044_ppc64el.deb ...
Unpacking xcat (2.12-snap201605052044) over (2.12-snap201604040933) ...
Processing triggers for ureadahead (0.100.0-19) ...
Processing triggers for systemd (229-4ubuntu4) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Processing triggers for install-info (6.1.0.dfsg.1-5) ...
Errors were encountered while processing:
 /xcatn/xcat-core/pool/main/x/xcat-genesis-scripts/xcat-genesis-scripts_2.12-snap201605052044_ppc64el.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@c910f03fsp03v120:/xcatn/xcat-core# 
```

**Comment by junxiawang** [comment]
```
apt-get upgrade:

Setting up xcat-server (2.12-snap201605052044) 
Reloading xcatd Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^config \* { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/conserver.pm line 194.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/rmdev -l \${ <-- HERE BASECUST_REMOVAL}/ at /opt/xcat/lib/perl/xCAT_plugin/aixinstall.pm line 7741.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^config \* { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/conserver.pm line 194.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/forwarders { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1063.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/allow-transfer { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1070.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/also-notify { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1077.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1084.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on-v6 { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1091.
root@c910f03fsp03v120:/xcatn/xcat-dep# perl -c /opt/xcat/lib/perl/xCAT_plugin/ddns.pm
Can't locate xCAT/Table.pm in @INC (you may need to install the xCAT::Table module) (@INC contains: /etc/perl /usr/local/lib/powerpc64le-linux-gnu/perl/5.22.1 /usr/local/share/perl/5.22.1 /usr/lib/powerpc64le-linux-gnu/perl5/5.22 /usr/share/perl5 /usr/lib/powerpc64le-linux-gnu/perl/5.22 /usr/share/perl/5.22 /usr/local/lib/site_perl /usr/lib/powerpc64le-linux-gnu/perl-base .) at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 6.
BEGIN failed--compilation aborted at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 6.
root@c910f03fsp03v120:/xcatn/xcat-dep# perl -I /opt/xcat/lib/perl/ -c /opt/xcat/lib/perl/xCAT_plugin/ddns.pm
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/forwarders { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1063.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/allow-transfer { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1070.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/also-notify { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1077.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1084.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/listen-on-v6 { <-- HERE / at /opt/xcat/lib/perl/xCAT_plugin/ddns.pm line 1091.
/opt/xcat/lib/perl/xCAT_plugin/ddns.pm syntax OK
root@c910f03fsp03v120:/xcatn/xcat-dep# 
```

**Comment by daniceexi** [comment]
This is a perl syntax warning, we can fix it in 2.12.1.

**Comment by junxiawang** [comment]
env:
ubuntu 16.04
